evaluate.go

```diff
@@ -237,10 +237,37 @@ func evaluateNotPresent(ptr pointerstructure.Pointer, datum interface{}) bool {
 	return reflect.ValueOf(val).Kind() == reflect.Map
 }
 
-func evaluateMatchExpression(expression *grammar.MatchExpression, datum interface{}, opt ...Option) (bool, error) {
+func getValue(datum interface{}, path []string, opt ...Option) (interface{}, bool, error) {
```

Comment:
It'd be nice to have a comment here just detailing what this function is doing (the overall flow with the local variables).

Reply:
Thanks, I added some comments now.